### PR TITLE
ipam: add Half

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

File: ipam.go [hunk 1]
```diff
@@ -182,3 +182,30 @@ func Free(network net.IPNet, mask net.IPMask, subnets []net.IPNet) (net.IPNet, e
 
 	return freeNetwork, nil
 }
+
+// Half takes a networks and returns two subnets which split this network in
```
Comment (JosephSalisbury) [inline ipam.go]: takes a network

File: ipam.go [hunk 2]
```diff
@@ -182,3 +182,30 @@ func Free(network net.IPNet, mask net.IPMask, subnets []net.IPNet) (net.IPNet, e
...
+// a half.
```
Comment (JosephSalisbury) [inline ipam.go]: split the network in half.

File: ipam.go [hunk 3]
```diff
@@ -182,3 +182,30 @@ func Free(network net.IPNet, mask net.IPMask, subnets []net.IPNet) (net.IPNet, e
...
+func Half(network net.IPNet) (first, second net.IPNet, err error) {
+	// One bit bigger mask will give a second half of a network.
```
Comment (JosephSalisbury) [inline ipam.go]: This comment doesn't sound right to me, but I couldn't give a better one :D

Comment (kopiczko) [reply ipam.go]: https://github.com/giantswarm/ipam/pull/4/files#diff-a6cb816e9597f944bea1a9807d8633efR194

File: ipam.go [hunk 4]
```diff
@@ -182,3 +182,30 @@ func Free(network net.IPNet, mask net.IPMask, subnets []net.IPNet) (net.IPNet, e
 
 	return freeNetwork, nil
 }
+
+// Half takes a network and returns two subnets which split the network in
```
Comment (xh3b4sd) [inline ipam.go]: `split` -> `splits`

Comment (JosephSalisbury) [reply ipam.go]: `split` is correct. The two networks split the larger network.

File: ipam.go (hunk 3 above)
Comment (JosephSalisbury) [reply ipam.go]: Grand

**Pull request merged**